### DISCUSSION TITLE
Creating custom visualizations as part of pipeline

### DISCUSSION THREAD

**Discussion by megha120** [apache/airflow#23616] [Q&A] [answered]
I am looking for a way to add a custom plot/visualization as part of a task. Is there any way we can create a plot and display it on the airflow ui (either on the dag details page or logs)?

**Comment by potiuk** [comment, marked as answer]
You can create custom Plugins with your own UI screens.. but this is not recommended way as the Web plugins will likely go away in Airflow 3. It's much better to use any tools you usually use for your data visualisation and retrieve data you want to visualise via Airlfow's REST API.

**Comment by megha120** [comment]
@potiuk Thanks for your response. I checked the custom plugins option, it looks like it can be integrated only using flask. Can we integrate streamlit app as a custom plugin to airflow?

**Reply by potiuk** [reply]
Why streamlit ? Why do you expect airflow to be able to support strramlit ?

**Reply by megha120** [reply]
We need to create some custom visualizations on streaming data

**Reply by potiuk** [reply]
Yeah. But what made you think it will support streamlit :) ? It has nothing to do with streamlit.
